### PR TITLE
Implement BlobStore configuration API changes

### PR DESCRIPTION
**Pull request by emerkle826**
* Update the BlobStore Interface names to be consistent with discussions on the mailing list
* Implement CRUD functions for BlobStoreInfo configurations.

### CODE REVIEW

File: geowebcache/core/src/main/java/org/geowebcache/config/BlobStoreConfiguration.java [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+
+    /**
+     * Replaces a {@link BlobStoreInfo}'s name with a new name.
+     * @param oldName The name that identifies the {@link BlobStoreInfo} to be renamed.
+     * @param newName The name that should replace the current name of the {@link BlobStoreInfo} identified by oldName.
+     * @throws NoSuchElementException If there is no BlobStore config identified by the provided oldName.
+     * @throws IllegalArgumentException If this configuration can't, for some reason, rename the blob store identified
+     * by the supplied name (for example, newName is null or invalid).
+     */
+    void renameBlobStore(String oldName, String newName) throws NoSuchElementException, IllegalArgumentException;
```
Comment (tbarsballe) [inline geowebcache/core/src/main/java/org/geowebcache/config/BlobStoreConfiguration.java]: Looks good. Looking at TileLayerConfiguration, a `containsBlobStore` method _might_ be valuable, but I'm not sure. `getBlobStore` is probably sufficient.

File: geowebcache/core/src/main/java/org/geowebcache/config/BlobStoreInfo.java [hunk 2]
```diff
@@ -70,6 +70,15 @@ public String getId() {
...
     }
 
+    /**
+     * Returns this {@link Info Info}s name. For now, this just returns this BlobStoreInfo's ID.
+     * @return A String representing the name of this Info implementation.
+     */
+    @Override
+    public String getName() {
+        return getId();
+    }
```
Comment (tbarsballe) [inline geowebcache/core/src/main/java/org/geowebcache/config/BlobStoreInfo.java]: Given we are switching to getName as the unique identifier, should also mark get/setId as `@Deprecated` (And probably also add a setName to replace setId)

Additionally, changing argument and variable names (mostly in the constructor) from id to name would probably be a good idea as well.

File: geowebcache/core/src/main/java/org/geowebcache/config/XMLConfiguration.java [hunk 3]
```diff
@@ -1086,10 +1086,186 @@ public Boolean getfullWMS(){
...
+        return Collections.unmodifiableList(getGwcConfig().getBlobStores().
+            stream().map((info) -> {
+                return (BlobStoreInfo)info.clone();
+            }).collect(Collectors.toList())
+        );
     }
-    
+
+    @Override
+    public synchronized void addBlobStore(BlobStoreInfo info) {
```
Comment (tbarsballe) [inline geowebcache/core/src/main/java/org/geowebcache/config/XMLConfiguration.java]: Similar to how TileLayerConfiguration methods are handled in XMLConfiguration, anything that overrides a BlobStoreConfiguration method should include a javadoc of the form:
```
/**
 * @see BlobStoreConfiguration#getBlobStores()
 */
```

**Comment by tbarsballe** [comment]
Update: After recent changes, this PR also has conflicts that must be resolved before it can be merged.

File: geowebcache/core/src/main/java/org/geowebcache/config/BlobStoreConfiguration.java (hunk 1 above)
Comment (emerkle826) [reply geowebcache/core/src/main/java/org/geowebcache/config/BlobStoreConfiguration.java]: @tbarsballe I've added 1`containsBlobStore` to be consistent.

Comment (emerkle826) [reply geowebcache/core/src/main/java/org/geowebcache/config/BlobStoreConfiguration.java]: added

**Comment by emerkle826** [comment]
I've merged the latest changes committed to the upstream branch and resolved conflicts.

**Pull request merged**